lib/common/utils/notification_service.dart

```diff
@@ -0,0 +1,90 @@
+import 'package:awesome_notifications/awesome_notifications.dart';
+import 'package:flutter/material.dart';
+import 'package:fluttercon/common/utils/router.dart';
+import 'package:fluttercon/core/theme/theme_colors.dart';
+import 'package:go_router/go_router.dart';
+import 'package:injectable/injectable.dart';
+import 'package:logger/logger.dart';
+
+@singleton
+class NotificationService {
+  factory NotificationService() {
+    return _notificationService;
+  }
+
+  NotificationService._internal();
+  static final NotificationService _notificationService =
+      NotificationService._internal();
```

Comment:
These would not be necessary since we have one instance available for injection as a singleton.